src/main/ruby/core/string.rb

```diff
@@ -1342,7 +1355,11 @@ def index(str, start=undefined)
       start = Rubinius::Type.coerce_to start, Fixnum, :to_int
 
       start += size if start < 0
-      return if start < 0 or start > size
+      if start < 0 or start > size
+        Truffle.invoke_primitive(:regexp_set_last_match, nil) if str.kind_of? Regexp
+
+        return
+      end
```

Comment:
Style: remove extra space?

Reply:
I really prefer `return`s to be separated, but if others don't I'll just adapt.

Reply:
To qualify that, I prefer them to be separated so it's very clear where the `return` statements are. And generally, I like statements grouped that logically go together. Unfortunately, that's more complex than Rubocop can handle.

Reply:
I agree with @bjfish here, it looks really odd to separate.

Reply:
Are there specs/tests for this? What if the Regexp has an incompatible encoding and the index are out of bounds?

Reply:
It was one of the untagged MRI tests.